tools/gen-postmortem-metadata.py

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env python
+
+#
+# gen-postmortem-metadata.py output_file.cc
+#
+# Creates debugging symbols to help naviage Node's internals using post-mortem
+# debugging tools.
+#
+
+import sys
+
+
+class DebugSymbol(object):
+  type_ = 'int'
+  _prefix = 'nodedbg_'
+
+  def __init__(self, name, value, headers=[], type_=None):
+    self.name = name
+    self.value = value
+    self.headers = headers
+    self.type_ = type_ or DebugSymbol.type_
+
+  @classmethod
+  def get_headers(cls, debug_symbols):
+    '''
+    Return a list of headers without duplicates, preserving the order they were
+    declared
+    '''
+    seen = set()
+    headers = [debug_symbol.headers for debug_symbol in debug_symbols]
+    headers = sum(headers, [])
+
+    result = []
+    for h in headers:
+      if not h in seen:
+        seen.add(h)
+        result.append(h)
+
+    return result
+
+  @property
+  def declare(self):
+    return '{type} {prefix}{name};'.format(
+      type=self.type_,
+      prefix=self._prefix,
+      name=self.name,
+    )
+
+  @property
+  def fill(self):
+    return '{prefix}{name} = {value};'.format(
+      prefix=self._prefix,
+      name=self.name,
+      value=self.value,
+    )
+
+
+debug_symbols = [
+  DebugSymbol(
+    name='environment_context_idx_embedder_data',
+    value='Environment::kContextEmbedderDataIndex',
+    headers=['env.h'],
+    type_='int',
+  ),
+  DebugSymbol(
+    name='class__BaseObject__persistent_handle',
+    value='offsetof(BaseObject, persistent_handle_)',
+    headers=['base-object.h', 'base-object-inl.h'],
```

Comment:
You could drop `base-object.h` here.  We recently instituted a rule that the `*-inl.h` is sufficient if you want the implementation (as opposed to just the declaration, which is what the stem makes available.)